app/models/compute_resources/foreman/model/ec2.rb

```diff
@@ -106,6 +106,27 @@ def image_exists?(image)
       client.images.get(image).present?
     end
 
+    def normalize_vm_attrs(vm_attrs)
+      normalized = slice_vm_attributes(vm_attrs, ['flavor_id', 'availability_zone', 'subnet_id', 'image_id', 'managed_ip'])
+
+      normalized['flavor_name'] =  self.flavors.detect { |f| f.id == normalized['flavor_id'] }.try(:name)
+      normalized['subnet_name'] =  self.subnets.detect { |f| f.subnet_id == normalized['subnet_id'] }.try(:cidr_block)
+      normalized['image_name'] = self.images.where(:uuid => vm_attrs['image_id']).first.try(:name)
+
+      group_ids = vm_attrs['security_group_ids'] || []
+      group_ids = group_ids.select { |gid| gid != '' }
+      normalized['security_groups'] = group_ids.map.with_index do |gid, idx|
+        [idx.to_s, {
+          'id' => gid,
+          'name' => self.security_groups.detect { |g| g.group_id == gid }.try(:name)
+        }]
+      end.to_h
+
+      normalized
+    rescue Fog::Compute::AWS::Error
```

Comment:
This exception should at least be logged.

Reply:
Ack, I'll add logging